core/notification/src/main/java/edu/stanford/spezi/core/notification/notifier/NotificationNotifier.kt

```diff
@@ -0,0 +1,93 @@
+package edu.stanford.spezi.core.notification.notifier
+
+import android.Manifest
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.content.PermissionChecker
+import dagger.hilt.android.qualifiers.ApplicationContext
+import edu.stanford.spezi.core.logging.speziLogger
+import edu.stanford.spezi.core.notification.R
+import javax.inject.Inject
+
+/**
+ * A class that sends notifications using the system tray.
+ */
+class NotificationNotifier @Inject constructor(
```

Comment:
Can we mark this as internal and move the message key in a companion object of `FirebaseMessage`?